### PR TITLE
[DEV-4192] root team

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: server/athenian/api/controllers/team_controller.py [hunk 1]
```diff
@@ -149,10 +163,18 @@ async def update_team(request: AthenianWebRequest, id: int,
                 raise ResponseError(BadRequestError(
                     detail="Team cannot be a the parent of itself.",
                 ))
-            meta_ids = await get_metadata_account_ids(account, sdb_conn, request.cache)
-            members = await _resolve_members(body.members, meta_ids, request.mdb)
+
             await _check_parent(account, body.parent, sdb_conn)
             await _check_parent_cycle(id, body.parent, sdb_conn)
+            meta_ids = await get_metadata_account_ids(account, sdb_conn, request.cache)
+            members = await _resolve_members(body.members, meta_ids, request.mdb)
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/team_controller.py:170]: Why did we reorder these checks?

File: server/athenian/api/internal/account.py [hunk 2]
```diff
@@ -347,12 +349,20 @@ async def copy_teams_as_needed(account: int,
...
         team = teams[node_id]
-        if (parent := teams.get(team[MetadataTeam.parent_team_id.name])) is not None:
-            parent = db_ids[parent[MetadataTeam.id.name]]
+
+        parent_id = None
+        if (meta_parent_id := team[MetadataTeam.parent_team_id.name]) is not None:
+            if (parent := teams.get(meta_parent_id)) is not None:
+                parent_id = db_ids[parent[MetadataTeam.id.name]]
+
+        if parent_id is None:
```
Comment (vmarkovtsev) [inline server/athenian/api/internal/account.py:358]:
```suggestion
        github_parent_id = team[MetadataTeam.parent_team_id.name]
        try:    
            parent_id = db_ids[teams[github_parent_id][MetadataTeam.id.name]]
        except KeyError:
            parent_id = root_team_id
```

File: server/athenian/api/models/state/models.py [hunk 3]
```diff
@@ -108,6 +108,7 @@ class Team(create_time_mixin(created_at=True, updated_at=True),
                       CheckConstraint("id != parent_id", name="cc_parent_self_reference"),
                       {"sqlite_autoincrement": True})
     BOTS = "Bots"  # the name of the special team which contains bots
+    ROOT = "Root"  # the name of the special artificial single root team
```
Comment (vmarkovtsev) [inline server/athenian/api/models/state/models.py:111]:
```suggestion
    ROOT = "Root"  # the name of the special artificial unique root team
```

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: server/athenian/api/internal/account.py (hunk 2 above)
Comment (gaetano-guerriero) [reply server/athenian/api/internal/account.py:358]: for me here it is less readable to try `teams[None]` when github team has no parent

File: server/athenian/api/controllers/team_controller.py (hunk 1 above)
Comment (gaetano-guerriero) [reply server/athenian/api/controllers/team_controller.py:170]: reverted

**Review by vmarkovtsev — APPROVED** [review]

File: server/athenian/api/internal/account.py (hunk 2 above)
Comment (vmarkovtsev) [reply server/athenian/api/internal/account.py]: I changed to this version in https://github.com/athenianco/athenian-api/pull/2392
```python
parent_id = root_team_id
if (github_parent_id := team[MetadataTeam.parent_team_id.name]) is not None:
    if (parent := teams.get(github_parent_id)) is not None:
        parent_id = db_ids[parent[MetadataTeam.id.name]]
```

**Pull request merged**